docs/TuxActions.md

```diff
@@ -0,0 +1,358 @@
+
+```
+    actionCategory.add.type === 'todos_add';
+```
+
+####<a id="actionCategory-before"></a>Method - `actionCategory.before` - type: FUNCTION [#](#actionCategory-before)
+Accepts an action verb string or array of action verbs and a callback, invokes the callback before dispatching the action when the action method is invoked.  The major use case for the `before` callback is to allow actions to invoke async functions (such as Ajax requests) and then dispatch the results of those requests.
+
+```
+    actionCategory.before(actionVerb, callbackToInvokeBeforeDispatching);
+```
+
+#####<a id="before-actionVerb"></a>Parameter - `actionVerb` - type: STRING or ARRAY of STRINGs - required [#](#before-actionVerb)
+Action (or array of actions) which will invoke the passed in callback before dispatching their corresponding actions.
+
+```
+    actionVerb = 'add'; //OR
+    actionVerb = ['add', 'remove'];
+```
+
+#####<a id="before-callbackToInvokeBeforeDispatching"></a>Parameter - `callbackToInvokeBeforeDispatching` - type: FUNCTION - required [#](#before-callbackToInvokeBeforeDispatching)
+Function to be invoked before the action to dispatch. Multiple `callbackToInvokeBeforeDispatching` callbacks can be registered to an action via the `before` method and they will be invoked in reverse order of how they are registered.  The last function to invoke will dispatch the action.  This chaining occurs by having each callback invoke the next callback in the series and, optionally, pass in an `actionBody` to it. This syntax allows for asynchronous callback chaining. The inputs to the callback are described below
+
+```
+    callbackToInvokeBeforeDispatching(nextCallbackInBeforeChain, actionBody);
+```
+
+######<a id="callbackToInvokeBeforeDispatching-nextCallbackInBeforeChain"></a>Parameter - `nextCallbackInBeforeChain` - type: FUNCTION [#](#callbackToInvokeBeforeDispatching-nextCallbackInBeforeChain)
+The next callback in the chain for this action.  The callback will receive whatever `actionBody` object is passed into it when invoked.  This allows chained callbacks to modify/replace submitted inputs.
+
+```
+    nextCallbackInBeforeChain(actionBody);
+```
+
+######<a id="callbackToInvokeBeforeDispatching-actionBody"></a>Parameter - `actionBody` - type: OBJECT [#](#callbackToInvokeBeforeDispatching-actionBody)
+This will either come from the input that the action was invoked with or the prior `before` callback that invoked this function.
+
+```
+    actionBody = {text: 'new body to pass into the next callback'};
+```
+
+Here is a complete example of the before method
+
+```
+    actionCategory.before('add', function (nextCallbackToInvokeBeforeDispatching, actionBody){
+      actionBody.text === 'text passed in from action or last before callback';
```

Comment:
Is there something missing here? Right now you're going to get a boolean without a conditional or a return.